addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -283,6 +284,46 @@ def test_on_next_order_reward_promotion_program_with_requirements(self):
         order_bis._update_programs_and_rewards()
         self.assertEqual(len(order_bis.order_line), 2, "Free product from a coupon generated from a promotion program on next order should not dissapear")
 
+    def test_partner_assigned_to_next_order_coupon(self):
+        # Test the assignment of a partner to a generated 'next order coupon'.
+        # Steps:
+        # 1. Create a loyalty program of type 'next_order_coupons'.
+        # 2. Create a sale order and add a product to it.
+        # 3. Apply the loyalty program to the sale order.
+        # 4. Verify that the generated coupon is assigned to the order's partner.
+
+        # 1.
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': '10% Discount on Next Order',
+            'program_type': 'next_order_coupons',
+            'applies_on': 'future',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {})],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 10,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+        # 2.
+        order = self.empty_order.copy()
+        order.write({'order_line': [
+            Command.create({
+                'product_id': self.product_A.id,
+                'name': '1 Product A',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            })
+        ]})
+        # 3.
+        result = order._try_apply_program(loyalty_program)
+        # 4.
+        generated_coupons = result.get('coupon')
+        self.assertTrue(generated_coupons, "A coupon should have been generated")
+        for coupon in generated_coupons:
```

Comment:
No need for that for loop even if `generated_coupons` contains multiple record because all coupons should be linked to one partner :thinking: 
Coupon: Partner (1, 5), (2, 5), (3, 5)
generated_coupons.partner_id will only return one partner (5)